core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java

```diff
@@ -164,7 +168,12 @@ public ExistingElementMutationImpl(T element) {
 
     @Override
     public ExistingElementMutation<T> alterPropertyVisibility(Property property, Visibility visibility) {
-        FetchHint.checkFetchHints(getElement().getFetchHints(), EnumSet.of(FetchHint.PROPERTIES, FetchHint.PROPERTY_METADATA));
+        if (!element.getFetchHints().isIncludeAllProperties()) {
```

Comment:
Same comment as above. Do we need all the properties or just the one you're altering?

Reply:
done